sql/rename.go

```diff
@@ -136,3 +136,57 @@ func (p *planner) RenameTable(n *parser.RenameTable) (planNode, error) {
 
 	return &valuesNode{}, nil
 }
+
+// RenameIndex renames the index.
+// Privileges: WRITE on database.
+//   Notes: postgres requires the table owner.
+//          mysql requires ALTER, CREATE, and INSERT on the table.
+func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
+	if n.NewName == "" {
+		return nil, errEmptyIndexName
+	}
+
+	if err := n.Name.NormalizeTableName(p.session.Database); err != nil {
+		return nil, err
+	}
+
+	if n.Name.Index() == string(n.NewName) {
+		// Noop.
+		return &valuesNode{}, nil
+	}
+
+	tableDesc, err := p.getTableDesc(n.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	idx := tableDesc.GetIndexByName(n.Name.Index())
+	if idx == nil {
+		if n.IfExists {
+			// Noop.
+			return &valuesNode{}, nil
+		}
+		// index does not exist, but we want it to: error out.
+		return nil, fmt.Errorf("index %q does not exist", n.Name.Index())
+	}
+
+	dbDesc, err := p.getDatabaseDesc(n.Name.Database())
+	if err != nil {
+		return nil, err
+	}
+	if err := p.checkPrivilege(dbDesc, privilege.WRITE); err != nil {
```

Comment:
Since index renaming is operating on the table descriptor, I would think you'd want to check the `tableDesc` privileges.